### ISSUE TITLE
Conan package

### ISSUE THREAD

**Issue by yliharsilamikko** [ocornut/imgui_test_engine#36]
Hello,

It would be very nice to have Conan package for imgui_test_engine as there already exists package for imgui itself.

ImGui Conan package issue (https://github.com/ocornut/imgui/issues/1469)
ImGui package in Conan center (https://conan.io/center/recipes/imgui)
ImGui package recipe (https://github.com/conan-io/conan-center-index/tree/master/recipes/imgui)

I find Conan very convenient tool for managing library dependencies. Fresh checkout can be easily build from scratch in fresh environment. Only requirements are that Conan + build tools are installed. Conan will automatically resolve package dependencies and load required packages into local machine.
 
Below are code pieces from my project using Conan+CMake for adding ImGui into project:
`
conanfile.py
	def requirements(self):
		self.requires("imgui/cci.20230105+1.89.2.docking")

CMakeLists.txt
	# find libraries
	find_package(imgui QUIET CONFIG)
	# link libraries
	target_link_libraries(${TARGET} PRIVATE imgui::imgui)

main.cpp
	#include "imgui.h"
	ImGui::CreateContext();
`

**Comment by ocornut** [comment]
I don’t see reason for the package to be hosted and maintained by us? Why not creating a conan imgui repository with that stuff ?

**Comment by yliharsilamikko** [comment]
> I don’t see reason for the package to be hosted and maintained by us? Why not creating a conan imgui repository with that stuff ?

You are right about hosting and maintaining. I created issuen conan-center repository (https://github.com/conan-io/conan-center-index/issues/20552)
Recipe would be hosted in conan-io/conan-center-index/tree/master/recipes. Creating and maintaining the recipe would require some effort. I could try to look if I can create the initial recipe.

If/when the Conan-recipe is added, it could be mentioned in wiki/Setting-Up

**Comment by jacobfriedman** [comment]
Hello all, been awhile! I can support maintaining that recipe but have to
update to the latest imgui in my local project. Could take a month.
Header-only makes it easier with the combination of requirements
(glfw/sdl/etc.) Not to forget metal.

The conan team is responsive and I'm sure the recipe will be welcomed.
Conan is useful!

**Comment by ocornut** [comment]
Closing this.
